### ISSUE TITLE
Recommended platform?

### ISSUE THREAD

**Issue by cvweiss** [sahat/hackathon-starter#523]
Installing from scratch, npm throws multiple errors as well as warns of multiple deprecations.

What platforms are recommended for running this bootstrap such that it'll work "out of the box"?

**Comment by sahat** [comment]
All platforms are supported. I would need to see your error log before I can help you with anything.

**Comment by cvweiss** [comment]
My apologies, was on vacation for a couple of days.

* Cloned using git
* ran npm install

Looks like it is actually building today, perhaps one of the libraries fixed an error. Still, the numerous deprecations gives me pause.

```
npm WARN deprecated jade@1.11.0: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN engine xmlbuilder@8.2.2: wanted: {"node":">=4.0"} (current: {"node":"0.12.15","npm":"2.15.1"})
npm WARN deprecated transformers@2.1.0: Deprecated, use jstransformer
npm WARN engine punycode@2.0.0: wanted: {"node":">=6"} (current: {"node":"0.12.15","npm":"2.15.1"})
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue

> secp256k1@3.0.1 install /var/www/test/node_modules/bitgo/node_modules/secp256k1
> npm run rebuild

|
> secp256k1@3.0.1 rebuild /var/www/test/node_modules/bitgo/node_modules/secp256k1
> node-gyp rebuild

gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
make: Entering directory '/var/www/test/node_modules/bitgo/node_modules/secp256k1/build'
  CXX(target) Release/obj.target/secp256k1/src/addon.o
  CXX(target) Release/obj.target/secp256k1/src/privatekey.o
  CXX(target) Release/obj.target/secp256k1/src/publickey.o
  CXX(target) Release/obj.target/secp256k1/src/signature.o
  CXX(target) Release/obj.target/secp256k1/src/ecdsa.o
  CXX(target) Release/obj.target/secp256k1/src/ecdh.o
  CC(target) Release/obj.target/secp256k1/src/secp256k1-src/src/secp256k1.o
  CC(target) Release/obj.target/secp256k1/src/secp256k1-src/contrib/lax_der_privatekey_parsing.o
  SOLINK_MODULE(target) Release/obj.target/secp256k1.node
  COPY Release/secp256k1.node
make: Leaving directory '/var/www/test/node_modules/bitgo/node_modules/secp256k1/build'

> node-sass@3.8.0 install /var/www/test/node_modules/node-sass-middleware/node_modules/node-sass
> node scripts/install.js

Binary downloaded and installed at /var/www/test/node_modules/node-sass-middleware/node_modules/node-sass/vendor/linux-x64-14/binding.node

> sha3@1.2.0 install /var/www/test/node_modules/bitgo/node_modules/keccakjs/node_modules/sha3
> node-gyp rebuild

gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
make: Entering directory '/var/www/test/node_modules/bitgo/node_modules/keccakjs/node_modules/sha3/build'
  CXX(target) Release/obj.target/sha3/src/addon.o
  CXX(target) Release/obj.target/sha3/src/displayIntermediateValues.o
  CXX(target) Release/obj.target/sha3/src/KeccakF-1600-reference.o
  CXX(target) Release/obj.target/sha3/src/KeccakNISTInterface.o
  CXX(target) Release/obj.target/sha3/src/KeccakSponge.o
  SOLINK_MODULE(target) Release/obj.target/sha3.node
  COPY Release/sha3.node
make: Leaving directory '/var/www/test/node_modules/bitgo/node_modules/keccakjs/node_modules/sha3/build'

> node-sass@3.8.0 postinstall /var/www/test/node_modules/node-sass-middleware/node_modules/node-sass
> node scripts/build.js

"/var/www/test/node_modules/node-sass-middleware/node_modules/node-sass/vendor/linux-x64-14/binding.node" exists.
 testing binary.
Binary is fine; exiting.
dotenv@2.0.0 node_modules/dotenv

bcrypt-nodejs@0.0.3 node_modules/bcrypt-nodejs

express-flash@0.0.2 node_modules/express-flash
└── connect-flash@0.1.1

passport-local@1.0.0 node_modules/passport-local
└── passport-strategy@1.0.0

passport@0.3.2 node_modules/passport
├── passport-strategy@1.0.0
└── pause@0.0.1

lusca@1.4.1 node_modules/lusca
└── core-util-is@1.0.2

tumblr.js@1.1.0 node_modules/tumblr.js
└── query-string@4.2.2 (strict-uri-encode@1.1.0, object-assign@4.1.0)

morgan@1.7.0 node_modules/morgan
├── on-headers@1.0.1
├── basic-auth@1.0.4
├── depd@1.1.0
├── on-finished@2.3.0 (ee-first@1.1.1)
└── debug@2.2.0 (ms@0.7.1)

express-session@1.14.0 node_modules/express-session
├── on-headers@1.0.1
├── utils-merge@1.0.0
├── cookie-signature@1.0.6
├── parseurl@1.3.1
├── cookie@0.3.1
├── depd@1.1.0
├── crc@3.4.0
├── uid-safe@2.1.1 (base64-url@1.2.2, random-bytes@1.0.0)
└── debug@2.2.0 (ms@0.7.1)

passport-openid@0.4.0 node_modules/passport-openid
├── passport-strategy@1.0.0
└── openid@1.0.0

validator@5.4.0 node_modules/validator

clockwork@0.1.4 node_modules/clockwork
├── xmlbuilder@8.2.2
└── xml2js@0.4.17 (sax@1.2.1, xmlbuilder@4.2.1)

lastfm@0.9.2 node_modules/lastfm
└── underscore@1.6.0

paypal-rest-sdk@1.6.9 node_modules/paypal-rest-sdk
├── buffer-crc32@0.2.5
├── semver@5.3.0
└── uuid@2.0.2

passport-twitter@1.0.4 node_modules/passport-twitter
├── xtraverse@0.1.0 (xmldom@0.1.22)
└── passport-oauth1@1.1.0 (utils-merge@1.0.0, passport-strategy@1.0.0, oauth@0.9.14)

passport-oauth@1.0.0 node_modules/passport-oauth
├── passport-oauth1@1.1.0 (utils-merge@1.0.0, passport-strategy@1.0.0, oauth@0.9.14)
└── passport-oauth2@1.3.0 (uid2@0.0.3, passport-strategy@1.0.0, oauth@0.9.14)

passport-github@1.1.0 node_modules/passport-github
└── passport-oauth2@1.3.0 (uid2@0.0.3, passport-strategy@1.0.0, oauth@0.9.14)

passport-facebook@2.1.1 node_modules/passport-facebook
└── passport-oauth2@1.3.0 (uid2@0.0.3, passport-strategy@1.0.0, oauth@0.9.14)

passport-linkedin-oauth2@1.4.1 node_modules/passport-linkedin-oauth2
├── underscore@1.8.3
└── passport-oauth2@1.3.0 (uid2@0.0.3, passport-strategy@1.0.0, oauth@0.9.14)

passport-instagram@1.0.0 node_modules/passport-instagram
└── passport-oauth2@1.3.0 (uid2@0.0.3, passport-strategy@1.0.0, oauth@0.9.14)

errorhandler@1.4.3 node_modules/errorhandler
├── escape-html@1.0.3
└── accepts@1.3.3 (negotiator@0.6.1, mime-types@2.1.11)

compression@1.6.2 node_modules/compression
├── on-headers@1.0.1
├── vary@1.1.0
├── bytes@2.3.0
├── debug@2.2.0 (ms@0.7.1)
├── compressible@2.0.8 (mime-db@1.23.0)
└── accepts@1.3.3 (negotiator@0.6.1, mime-types@2.1.11)

passport-google-oauth@1.0.0 node_modules/passport-google-oauth
├── passport-google-oauth1@1.0.0 (passport-oauth1@1.1.0)
└── passport-google-oauth20@1.0.0 (passport-oauth2@1.3.0)

instagram-node@0.5.8 node_modules/instagram-node
├── colors@0.6.0-1
└── fwk@1.0.11 (async@0.2.10, should@0.6.3)

express@4.14.0 node_modules/express
├── escape-html@1.0.3
├── utils-merge@1.0.0
├── cookie-signature@1.0.6
├── methods@1.1.2
├── vary@1.1.0
├── parseurl@1.3.1
├── array-flatten@1.1.1
├── content-type@1.0.2
├── etag@1.7.0
├── merge-descriptors@1.0.1
├── fresh@0.3.0
├── path-to-regexp@0.1.7
├── encodeurl@1.0.1
├── range-parser@1.2.0
├── cookie@0.3.1
├── content-disposition@0.5.1
├── serve-static@1.11.1
├── depd@1.1.0
├── on-finished@2.3.0 (ee-first@1.1.1)
├── finalhandler@0.5.0 (unpipe@1.0.0, statuses@1.3.0)
├── qs@6.2.0
├── debug@2.2.0 (ms@0.7.1)
├── proxy-addr@1.1.2 (forwarded@0.1.0, ipaddr.js@1.1.1)
├── send@0.14.1 (destroy@1.0.4, ms@0.7.1, statuses@1.3.0, mime@1.3.4, http-errors@1.5.0)
├── type-is@1.6.13 (media-typer@0.3.0, mime-types@2.1.11)
└── accepts@1.3.3 (negotiator@0.6.1, mime-types@2.1.11)

body-parser@1.15.2 node_modules/body-parser
├── bytes@2.4.0
├── content-type@1.0.2
├── depd@1.1.0
├── on-finished@2.3.0 (ee-first@1.1.1)
├── qs@6.2.0
├── debug@2.2.0 (ms@0.7.1)
├── raw-body@2.1.7 (unpipe@1.0.0)
├── http-errors@1.5.0 (setprototypeof@1.0.1, inherits@2.0.1, statuses@1.3.0)
├── type-is@1.6.13 (media-typer@0.3.0, mime-types@2.1.11)
└── iconv-lite@0.4.13

chai@3.5.0 node_modules/chai
├── assertion-error@1.0.2
├── type-detect@1.0.0
└── deep-eql@0.1.3 (type-detect@0.1.1)

async@2.0.0 node_modules/async

nodemailer@2.5.0 node_modules/nodemailer
├── socks@1.1.9 (ip@1.1.3, smart-buffer@1.0.11)
├── nodemailer-shared@1.0.5 (nodemailer-fetch@1.4.0)
├── mailcomposer@3.10.0 (buildmail@3.8.0)
├── nodemailer-smtp-transport@2.6.0 (nodemailer-wellknown@0.1.10, smtp-connection@2.8.0)
├── nodemailer-smtp-pool@2.7.0 (nodemailer-wellknown@0.1.10, smtp-connection@2.8.0)
├── nodemailer-direct-transport@3.2.0 (smtp-connection@2.8.0)
└── libmime@2.0.3 (libbase64@0.1.0, libqp@1.1.0, iconv-lite@0.4.13)

lob@3.9.0 node_modules/lob
└── bluebird@3.4.1

express-validator@2.20.8 node_modules/express-validator
└── bluebird@3.4.1

supertest@1.2.0 node_modules/supertest
├── methods@1.1.2
└── superagent@1.8.3 (component-emitter@1.2.1, cookiejar@2.0.6, reduce-component@1.0.1, extend@3.0.0, mime@1.3.4, formidable@1.0.17, qs@2.3.3, debug@2.2.0, readable-stream@1.0.27-1, form-data@1.0.0-rc3)

request@2.73.0 node_modules/request
├── is-typedarray@1.0.0
├── aws-sign2@0.6.0
├── forever-agent@0.6.1
├── caseless@0.11.0
├── stringstream@0.0.5
├── oauth-sign@0.8.2
├── tunnel-agent@0.4.3
├── isstream@0.1.2
├── json-stringify-safe@5.0.1
├── extend@3.0.0
├── aws4@1.4.1
├── node-uuid@1.4.7
├── combined-stream@1.0.5 (delayed-stream@1.0.0)
├── qs@6.2.0
├── mime-types@2.1.11 (mime-db@1.23.0)
├── tough-cookie@2.2.2
├── form-data@1.0.0-rc4 (async@1.5.2)
├── bl@1.1.2 (readable-stream@2.0.6)
├── har-validator@2.0.6 (pinkie-promise@2.0.1, commander@2.9.0, chalk@1.1.3, is-my-json-valid@2.13.1)
├── http-signature@1.1.1 (assert-plus@0.2.0, jsprim@1.3.0, sshpk@1.8.3)
└── hawk@3.1.3 (cryptiles@2.0.5, sntp@1.0.9, boom@2.10.1, hoek@2.16.3)

node-linkedin@0.5.3 node_modules/node-linkedin
├── oauth@0.9.14
├── request@2.31.0 (aws-sign2@0.5.0, forever-agent@0.5.2, qs@0.6.6, tunnel-agent@0.3.0, oauth-sign@0.3.0, json-stringify-safe@5.0.1, mime@1.2.11, node-uuid@1.4.7, form-data@0.1.4, http-signature@0.10.1, tough-cookie@0.9.15, hawk@1.0.0)
└── lodash@2.4.2

node-foursquare@0.3.0 node_modules/node-foursquare
└── winston@0.6.2 (cycle@1.0.3, stack-trace@0.0.9, eyes@0.1.8, colors@0.6.2, pkginfo@0.2.3, async@0.1.22, request@2.9.203)

fbgraph@1.2.0 node_modules/fbgraph
├── qs@1.2.2
└── request@2.72.0 (is-typedarray@1.0.0, oauth-sign@0.8.2, aws-sign2@0.6.0, forever-agent@0.6.1, tunnel-agent@0.4.3, caseless@0.11.0, stringstream@0.0.5, isstream@0.1.2, json-stringify-safe@5.0.1, extend@3.0.0, aws4@1.4.1, combined-stream@1.0.5, node-uuid@1.4.7, qs@6.1.0, tough-cookie@2.2.2, mime-types@2.1.11, form-data@1.0.0-rc4, bl@1.1.2, har-validator@2.0.6, http-signature@1.1.1, hawk@3.1.3)

stripe@4.8.0 node_modules/stripe
├── qs@2.4.2
└── bluebird@2.10.2

mocha@2.5.3 node_modules/mocha
├── escape-string-regexp@1.0.2
├── supports-color@1.2.0
├── growl@1.9.2
├── to-iso-string@0.0.2
├── commander@2.3.0
├── diff@1.4.0
├── debug@2.2.0 (ms@0.7.1)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── glob@3.2.11 (inherits@2.0.1, minimatch@0.3.0)
└── jade@0.26.3 (commander@0.6.1, mkdirp@0.3.0)

multer@1.1.0 node_modules/multer
├── object-assign@3.0.0
├── xtend@4.0.1
├── append-field@0.1.0
├── on-finished@2.3.0 (ee-first@1.1.1)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── type-is@1.6.13 (media-typer@0.3.0, mime-types@2.1.11)
├── concat-stream@1.5.1 (inherits@2.0.1, typedarray@0.0.6, readable-stream@2.0.6)
└── busboy@0.2.13 (readable-stream@1.1.14, dicer@0.2.5)

twilio@3.3.0-edge node_modules/twilio
├── deprecate@0.1.0
├── string.prototype.startswith@0.2.0
├── scmp@0.0.3
├── jwt-simple@0.3.1
├── underscore@1.8.3
├── q@0.9.7
├── jsonwebtoken@5.4.1 (ms@0.7.1, jws@3.1.3)
└── request@2.55.0 (caseless@0.9.0, aws-sign2@0.5.0, forever-agent@0.6.1, tunnel-agent@0.4.3, stringstream@0.0.5, oauth-sign@0.6.0, isstream@0.1.2, json-stringify-safe@5.0.1, node-uuid@1.4.7, qs@2.4.2, combined-stream@0.0.7, form-data@0.2.0, mime-types@2.0.14, tough-cookie@2.2.2, http-signature@0.10.1, bl@0.9.5, hawk@2.3.1, har-validator@1.8.0)

twit@2.2.4 node_modules/twit
├── mime@1.3.4
├── bluebird@3.4.1
└── request@2.58.0 (caseless@0.10.0, aws-sign2@0.5.0, oauth-sign@0.8.2, forever-agent@0.6.1, tunnel-agent@0.4.3, stringstream@0.0.5, isstream@0.1.2, json-stringify-safe@5.0.1, extend@2.0.1, combined-stream@1.0.5, node-uuid@1.4.7, qs@3.1.0, mime-types@2.0.14, tough-cookie@2.2.2, http-signature@0.11.0, bl@0.9.5, form-data@1.0.0-rc4, hawk@2.3.1, har-validator@1.8.0)

jade@1.11.0 node_modules/jade
├── character-parser@1.2.1
├── void-elements@2.0.1
├── commander@2.6.0
├── mkdirp@0.5.1 (minimist@0.0.8)
├── jstransformer@0.0.2 (is-promise@2.1.0, promise@6.1.0)
├── constantinople@3.0.2 (acorn@2.7.0)
├── with@4.0.3 (acorn-globals@1.0.9, acorn@1.2.2)
├── clean-css@3.4.18 (commander@2.8.1, source-map@0.4.4)
├── transformers@2.1.0 (promise@2.0.0, css@1.0.8, uglify-js@2.2.5)
└── uglify-js@2.7.0 (uglify-to-browserify@1.0.2, async@0.2.10, yargs@3.10.0, source-map@0.5.6)

connect-mongo@1.2.1 node_modules/connect-mongo
├── bluebird@3.4.1
└── mongodb@2.1.21 (es6-promise@3.0.2, readable-stream@1.0.31, mongodb-core@1.3.21)

mongoose@4.5.4 node_modules/mongoose
├── ms@0.7.1
├── sliced@1.0.1
├── regexp-clone@0.0.1
├── muri@1.1.0
├── hooks-fixed@1.1.0
├── mpromise@0.5.5
├── kareem@1.1.3
├── mpath@0.2.1
├── async@1.5.2
├── bson@0.4.23
├── mquery@1.11.0 (sliced@0.0.5, debug@2.2.0, bluebird@2.10.2)
└── mongodb@2.1.21 (es6-promise@3.0.2, readable-stream@1.0.31, mongodb-core@1.3.21)

cheerio@0.20.0 node_modules/cheerio
├── dom-serializer@0.1.0 (domelementtype@1.1.3)
├── entities@1.1.1
├── css-select@1.2.0 (boolbase@1.0.0, css-what@2.1.0, nth-check@1.0.1, domutils@1.5.1)
├── htmlparser2@3.8.3 (domelementtype@1.3.0, domutils@1.5.1, entities@1.0.0, domhandler@2.3.0, readable-stream@1.1.14)
└── jsdom@7.2.2 (acorn-globals@1.0.9, webidl-conversions@2.0.1, abab@1.0.3, xml-name-validator@2.0.1, sax@1.2.1, cssom@0.3.1, symbol-tree@3.1.4, nwmatcher@1.3.8, tough-cookie@2.2.2, whatwg-url-compat@0.6.5, parse5@1.5.1, acorn@2.7.0, escodegen@1.8.0, cssstyle@0.2.36)

lodash@4.13.1 node_modules/lodash

github-api@2.3.0 node_modules/github-api
├── utf8@2.1.1
├── js-base64@2.1.9
├── debug@2.2.0 (ms@0.7.1)
├── es6-promise@3.2.1
└── axios@0.10.0 (follow-redirects@0.0.7)

yui@3.18.1 node_modules/yui
└── request@2.40.0 (forever-agent@0.5.2, aws-sign2@0.5.0, tunnel-agent@0.4.3, oauth-sign@0.3.0, stringstream@0.0.5, json-stringify-safe@5.0.1, qs@1.0.2, node-uuid@1.4.7, mime-types@1.0.2, form-data@0.1.4, tough-cookie@2.2.2, http-signature@0.10.1, hawk@1.1.1)

bitgo@2.0.0 node_modules/bitgo
├── bs58@2.0.1
├── bs58check@1.0.4
├── ecurve@1.0.3
├── big.js@3.1.3
├── bigi@1.4.0
├── underscore.string@2.4.0
├── minimist@0.2.0
├── q@1.1.2
├── http-proxy@1.11.1 (eventemitter3@1.2.0, requires-port@0.0.1)
├── morgan@1.5.3 (basic-auth@1.0.4, depd@1.0.1, on-finished@2.2.1, debug@2.2.0)
├── assert@0.4.9 (util@0.10.3)
├── create-hmac@1.1.4 (inherits@2.0.1, create-hash@1.1.2)
├── argparse@0.1.16 (underscore@1.7.0)
├── ethereumjs-abi@0.6.2 (bn.js@4.11.5)
├── ethereumjs-util@4.5.0 (rlp@2.0.0, create-hash@1.1.2, bn.js@4.11.5)
├── bitcoinjs-lib@2.1.4 (buffer-equals@1.0.3, buffer-reverse@1.0.1, bip66@1.1.4, randombytes@2.0.3, wif@1.2.1, bs58check@1.0.8, create-hash@1.1.2, typeforce@1.6.2)
├── superagent@1.8.3 (methods@1.1.2, component-emitter@1.2.1, cookiejar@2.0.6, reduce-component@1.0.1, extend@3.0.0, mime@1.3.4, formidable@1.0.17, qs@2.3.3, debug@2.2.0, readable-stream@1.0.27-1, form-data@1.0.0-rc3)
├── superagent-proxy@1.0.1 (debug@2.2.0, proxy-agent@2.0.0)
├── moment@2.14.1
├── secp256k1@3.0.1 (bindings@1.2.1, bn.js@4.11.5, elliptic@6.3.1, drbg.js@1.0.1, nan@2.4.0)
└── keccakjs@0.2.1 (browserify-sha3@0.0.1, sha3@1.2.0)

node-sass-middleware@0.9.8 node_modules/node-sass-middleware
├── mkdirp@0.5.1 (minimist@0.0.8)
└── node-sass@3.8.0 (get-stdin@4.0.1, async-foreach@0.1.3, in-publish@2.0.0, chalk@1.1.3, lodash.clonedeep@4.3.2, cross-spawn@3.0.1, glob@7.0.5, nan@2.4.0, meow@3.7.0, sass-graph@2.1.2, gaze@1.1.0, node-gyp@3.4.0)
```
